rules/Renaming/Rector/FunctionLike/RenameFunctionLikeParamWithinCallLikeArgRector.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Renaming\Rector\FunctionLike;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\FunctionLike;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Param;
+use PhpParser\NodeTraverser;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Rector\AbstractRector;
+use Rector\Renaming\ValueObject\RenameFunctionLikeParamWithinCallLikeArg;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Tests\Renaming\Rector\FunctionLike\RenameFunctionLikeParamWithinCallLikeArgRector\RenameFunctionLikeParamWithinCallLikeArgRectorTest
+ */
+final class RenameFunctionLikeParamWithinCallLikeArgRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var RenameFunctionLikeParamWithinCallLikeArg[]
+     */
+    private array $renameFunctionLikeParamWithinCallLikeParams = [];
+
+    private bool $hasChanged = false;
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add param types where needed', [
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+(new SomeClass)->process(function ($param) {});
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+(new SomeClass)->process(function ($parameter) {});
+CODE_SAMPLE
+                ,
+                [new RenameFunctionLikeParamWithinCallLikeArg('SomeClass', 'process', 0, 0, 'parameter')]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param CallLike $node
+     */
+    public function refactor(Node $node): ?Node
+    {
```

Comment:
Like other PR, `$this->hasChanged = false` needs init here:

```suggestion
    {
        $this->hasChanged = false;
```